externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkArrowbasedOperationSuite.scala

```diff
@@ -138,6 +144,85 @@ class SparkArrowbasedOperationSuite extends WithSparkSQLEngine with SparkDataTyp
     assert(metrics("numOutputRows").value === 1)
   }
 
+  test("SparkDatasetHelper.executeArrowBatchCollect should return expect row count") {
+    val returnSize = Seq(
+      7, // less than one partition
+      10, // equal to one partition
+      13, // between one and two partitions, run two jobs
+      20, // equal to two partitions
+      29) // between two and three partitions
+
+    // aqe
```

Comment:
is there somewhere tune on the config AQE to make sure it is applied ?

Reply:
we have assertion `assert(headPlan.isInstanceOf[AdaptiveSparkPlanExec])` in L169, is that enough?

Reply:
added new ut `aqe should work properly`